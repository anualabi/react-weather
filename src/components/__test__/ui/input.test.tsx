import { render, screen } from "@testing-library/react";

import Input, { InputProps } from "@/components/ui/input";

const mockProps = {
  type: "text",
  placeholder: "Enter city name",
  className: "",
};

const customRender = (props: InputProps) => {
  return render(<Input {...props} />);
};

describe("Input", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = customRender({ ...mockProps });
    expect(asFragment()).toMatchSnapshot();
  });

  it("shold render an input HTML element", () => {
    customRender({ ...mockProps });
    const inputElement = screen.getByRole("textbox");
    expect(inputElement.tagName).toBe("INPUT");
  });

  it("should display the correct placeholder text", () => {
    render(<Input {...mockProps} />);
    const inputElement = screen.getByPlaceholderText(mockProps.placeholder);
    expect(inputElement).toBeInTheDocument();
  });
});
