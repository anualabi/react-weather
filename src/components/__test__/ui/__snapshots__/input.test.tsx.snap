// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render (snapshot) 1`] = `
<DocumentFragment>
  <input
    class="flex h-12 w-full rounded-md border border-black bg-secondary px-3 py-1 text-sm text-white shadow-sm transition-colors placeholder:text-zinc-500 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-zinc-950 disabled:cursor-not-allowed disabled:opacity-50 lg:text-base"
    placeholder="Enter city name"
    type="text"
  />
</DocumentFragment>
`;
